### DISCUSSION TITLE
Will apps that auto update ever have their (brew tracked) version incremented?

### DISCUSSION THREAD

**Discussion by IdyllicHappiness** [Homebrew/discussions#3605] [Casks]
After getting a MacBook in the past year I've been using Homebrew a lot, and been very impressed with it!

This post is very much just for my education rather than implying a problem or alike, I fully understand the need for the `auto_updates` property and its associated effects (unless `greedy` is specified, allow the app to update itself), but I'm curious:

- 1. Will the version as listed through Homebrew will be _forever_ static, keeping the version number it had at its initial install?

And further to this, I'm curious about the implications it has in terms of a package manager ecosystem.

With non-casks, I imagine Homebrew works much like Linux package managers, in that packages can be dependant on each other. If package-A is on version 1.23, relying on package-B 4.56, then package-B releases a vitally important security fix with 4.57, package-A can update to 1.24 with an updated dependency to reference the latest package-B.

 - 2. Is this the case?
 - 3. If this is the case, are dependants "forced" to update when the dependancy updates, or can they references older versions? (e.g. could the ecosystem be in a state such that package-A depends on B-4.56, and package-C depend on B-4.57?)

And with casks given they auto update with artifacts they download themselves, and their Homebrew tracked version number might not be accurate to the real state of the app, then these packages are completely isolated, and don't tend to have dependancies or dependants.

 - 4. Is this also the case?

As mentioned at the top of the post, this is not veiled critique or anything, just trying to learn more about how the ecosystem works 🙂 Cheers!

**Comment by SMillerDev** [comment]
> Will the version as listed through Homebrew will be forever static, keeping the version number it had at its initial install?

If you don't update it, yes.

> Is this the case?
> If this is the case, are dependants "forced" to update when the dependancy updates, or can they references older versions? (e.g. could the ecosystem be in a state such that package-A depends on B-4.56, and package-C depend on B-4.57?)

It can only happen on accident since generally we reference dependencies as just B. So the version isn't part of the dependency.

> And with casks given they auto update with artifacts they download themselves, and their Homebrew tracked version number might not be accurate to the real state of the app, then these packages are completely isolated, and don't tend to have dependancies or dependants.

You could define a cask dependency. The reason homebrew generally doesn't is because casks are self contained so don't need other casks. And formula should only depend on software that is open source.
